### PR TITLE
Implement nancumsum function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nancumsum, which computes the cumulative sum of elements along a specified axis while ignoring NaN values in the input tensor (i.e., treating NaNs as zero during accumulation).

The operation follows NumPy semantics for axis and dtype, and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/ops/numpy_test.py [hunk 1]
```diff
@@ -1716,6 +1716,23 @@ def test_moveaxis(self):
...
+        self.assertEqual(knp.nancumsum(x, axis=(1,)).shape, (None, 3, 3))
+        self.assertEqual(knp.nancumsum(x, axis=()).shape, (None, 3, 3))
+
+        self.assertEqual(knp.nancumsum(x, axis=None).shape, (None,))
+
+        x4 = KerasTensor((None, 2, 3, 4))
+
+        self.assertEqual(knp.nancumsum(x4, axis=2).shape, (None, 2, 3, 4))
+        self.assertEqual(knp.nancumsum(x4, axis=(1, 3)).shape, (None, 2, 3, 4))
+
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy_test.py:1730]: Updated.
